enterprise_scheduler/executor.py

```diff
@@ -85,30 +84,31 @@ def execute_task(self, task):
 
         ffdl_endpoint = task['endpoint']
         ffdl_authorization = task['user']
+        ffdl_authorization_pass = "temporary"
         ffdl_userinfo = task['userinfo']
         ffdl_zip = self._create_ffdl_zip(task)
         ffdl_manifest = self._create_manifest(task)
+        ffdl_ui_port = "32150"  ## FFDL UI hosting can vary
 
-        command = 'curl -X POST "' + ffdl_endpoint + '" ' \
-                  '-H "accept: application/json" ' \
-                  '-H "Authorization: ' + ffdl_authorization + '" ' \
-                  '-H "X-Watson-Userinfo: ' + ffdl_userinfo + '" ' \
-                  '-H "Content-Type: multipart/form-data" ' \
-                  '-F "model_definition=@' + ffdl_zip + ';type=application/zip" ' \
-                  '-F "manifest=@' + ffdl_manifest + '"'
+        task_headers = {"X-Watson-Userinfo": ffdl_userinfo}
 
-        #print('>>> Submitting \n {}'.format(command))
+        task_files = {'model_definition': open(ffdl_zip, 'rb'),
+                      "manifest": open(ffdl_manifest, 'rb')}
 
         try:
-            p = Popen([command ], stdout=PIPE, stderr=PIPE, shell=True, cwd=self.runtimedir)
-            p.wait()
-            #print(p.stdout.read())
-            #print(p.stderr.read())
-        finally:
-            if(p.stderr):
-                p.stderr.close()
-            if(p.stdout):
-                p.stdout.close()
+            result = requests.post(ffdl_endpoint,
+                                   auth=HTTPBasicAuth(ffdl_authorization, ffdl_authorization_pass),
+                                   headers=task_headers,
+                                   files=task_files)
+
+            print("FFDL API responded with status {} and response {}".format(result.status_code,
+                                                                             json.loads(result.content)))
+            #print("Training URL : http://{}:{}/#/trainings/{}".format(urlparse(ffdl_endpoint).netloc.split(":")[0],
```

Comment:
so, today we are displaying the console on the UI, does this provides us a URL that can take directly to the specific job being run (where the logs are visible)? If so, should we return that via the result to the client so it can be used on the UI?

Reply:
Yes this should output the provide a link your model in ffdl, still trying to get it working since it currently just brings to a blank page in the ui. I think we should pass this back to the little pop up notification after job submission, when its figured out.
Since the FFDL service can be deployed on any system(not necessarily the same system as the core services) we might need to add some sort of ['ffdl_ui_url'] field to construct the url needed for the client.